typing/typeclass.ml

```diff
@@ -827,7 +833,7 @@ and class_structure cl_num final val_env met_env loc
   let private_self = if final then Ctype.newvar () else self_type in
 
   (* Self binder *)
-  let (pat, meths, vars, val_env, meth_env, par_env) =
+  let (pat, meths, vars, val_env, met_env, par_env) =
     type_self_pattern cl_num private_self val_env met_env par_env spat
```

Comment:
Good question!

`type_self_pattern` could be changed to take a `class_env` parameter, and the return tuple could contain a `class_env` value instead of the three separate elements. But this function is defined in `typecore.ml`, not `typeclass.ml`, so doing this change would require moving the definition of the `class_env` type somewhere else (probably in typecore.ml, and then exposed publicly in typecore.mli), so that it can be seen from both modules.

The present PR is almost good-to-merge (there is a minor change to do on the Changes file to please our check-typo script). If you want to work on this (slightly more invasive) changed, I would recommend getting the present PR merged now without it, and then sending a follow-up PR with the change -- and other further improvements that you would like to consider, if any.

Reply:
@gasche  
Made the changes now all checks pass.
Okay I ll send a PR later about that change. :smile: 
Thanks